### PR TITLE
Group preregistration now works

### PR DESCRIPTION
**Pull request by EliAndrewC**
I need to do a bunch more testing, but I've fixed the stuff that was stopping group preregs from being possible.

This was all tested originally before I launched Magstock prereg, but since Magstock didn't use groups, it ended up breaking when I made further refactors.

I'll explain my changes inline.

### CODE REVIEW

File: development-defaults.ini [hunk 1]
```diff
@@ -151,10 +151,6 @@ no_shirt     = 0
 
 [enums]
```
Comment (EliAndrewC) [inline development-defaults.ini]: No reason to make these part of the core repo; we can add them in ``development.ini`` if a particular event needs that.

File: uber/models.py [hunk 2]
```diff
@@ -870,12 +870,7 @@ def floating(self):
 
     @property
     def new_ribbon(self):
-        ribbons = {a.ribbon for a in self.attendees}
-        for ribbon in [c.BAND_RIBBON]:
-            if ribbon in ribbons:
-                return ribbon
-        else:
-            return c.DEALER_ASST_RIBBON if self.is_dealer else c.NO_RIBBON
+        return c.DEALER_RIBBON if self.is_dealer else c.NO_RIBBON
```
Comment (EliAndrewC) [inline uber/models.py]: The above logic can now be added in a plugin for events which need it.

File: uber/models.py [hunk 3]
```diff
@@ -996,7 +991,7 @@ class Attendee(MagModel, TakesPaymentMixin):
     admin_notes = Column(UnicodeText, admin_only=True)
 
     badge_num  = Column(Integer, default=0, nullable=True, admin_only=True)
-    badge_type = Column(Choice(c.BADGE_OPTS), default=c.ATTENDEE_BADGE, admin_only=True)
+    badge_type = Column(Choice(c.BADGE_OPTS), default=c.ATTENDEE_BADGE)
```
Comment (EliAndrewC) [inline uber/models.py]: I forgot that we actually do let the attendee set the badge type in the prereg form, though obviously we check to make sure that someone didn't monkey with the form to give themselves a Guest badge or something.  Therefore this field is not ``admin_only``.

File: uber/templates/preregistration/form.html [hunk 4]
```diff
@@ -2,6 +2,13 @@
 {% block title %}Preregistration{% endblock %}
 {% block content %}
```
Comment (EliAndrewC) [inline uber/templates/preregistration/form.html]: This simplifies and fixes the spacing divs.

File: uber/templates/preregistration/form.html [hunk 5]
```diff
@@ -48,8 +56,10 @@
             $('input[name="badge_type"]').val('{{ c.ATTENDEE_BADGE }}');
             $('.badge_type_selector').removeClass('active');
             $('.badge_type_selector').slice(badgeTypeIndex, 1 + badgeTypeIndex).addClass('active');
-            $.field('amount_extra').val(badgeType.extra || 0).trigger('change');
             (badgeType.onClick || $.noop)();
+            if (badgeType.extra !== null) {
+                $.field('amount_extra').val(badgeType.extra || 0).trigger('change');
+            }
```
Comment (EliAndrewC) [inline uber/templates/preregistration/form.html]: This was the other problem with group prereg.  Triggering the ``change`` event caused the amount to be reset back to zero, which itself caused the badge type to go back to Attendee.  Whoops!

File: development-defaults.ini (hunk 1 above)
Comment (kitsuta) [inline development-defaults.ini]: Are you sure this wasn't part of some special-casing? I think these were left in because they prevented 500 errors.

Comment (EliAndrewC) [inline development-defaults.ini]: I did a grep of the codebase for all cases of ``_RIBBON`` and I didn't find anything other than the ones which are in configspec.ini and I just grepped specifically for ``INDIE_RIBBON`` and ``BAND_RIBBON`` in the code and didn't find anything (there are a couple of unit tests which reference them, but that's it and those tests are broken anyway :()

**Comment by kitsuta** [comment]
I code-reviewed this, I just gave the okay in Slack first. :+1:

**Pull request merged**